### PR TITLE
Story #576 combine transfer position topics

### PR DESCRIPTION
**Pull request by shashi165**
### Changes for [Story #576](https://github.com/mojaloop/project/issues/576)
1. combined the position topics
2. disable the cron job for registering position consumers

###Tests

```
=============================== Coverage summary ===============================
Statements   : 100% ( 3040/3040 )
Branches     : 100% ( 736/736 )
Functions    : 100% ( 426/426 )
Lines        : 100% ( 3012/3012 )
================================================================================

1..1111
# tests 1111
# pass  1111

# ok

```

### CODE REVIEW

File: src/handlers/lib/utility.js [hunk 1]
```diff
@@ -399,10 +399,10 @@ const createParticipantTopicConf = (participantName, functionality, action, part
  *
  * @returns {object} - Returns newly created general topicConfig
  */
-const createGeneralTopicConf = (functionality, action, partition = 0, opaqueKey = 0) => {
+const createGeneralTopicConf = (functionality, action, partition = 0, opaqueKey = Uuid()) => {
```
Comment (rmothilal) [inline src/handlers/lib/utility.js]: please add "key" parameter which defaults to null.
the opaqueKey should default to null

File: src/handlers/lib/utility.js [hunk 2]
```diff
@@ -399,10 +399,10 @@ const createParticipantTopicConf = (participantName, functionality, action, part
...
   return {
     topicName: transformGeneralTopicName(functionality, action),
-    key: Uuid(),
+    key: opaqueKey,
```
Comment (rmothilal) [inline src/handlers/lib/utility.js]: map key to passed in param

File: src/handlers/lib/utility.js [hunk 3]
```diff
@@ -426,15 +426,15 @@ const createGeneralTopicConf = (functionality, action, partition = 0, opaqueKey
  *
  * @returns {object} - Returns a boolean: true if successful, or throws and error if failed
  */
-const produceGeneralMessage = async (functionality, action, message, state) => {
+const produceGeneralMessage = async (functionality, action, message, state, participantName) => {
```
Comment (rmothilal) [inline src/handlers/lib/utility.js]: participantName should be key

File: src/handlers/positions/handler.js [hunk 4]
```diff
@@ -248,54 +247,28 @@ const positions = async (error, messages) => {
...
-      return false
+    const positionHandler = {
+      command: positions,
+      topicName: Utility.transformGeneralTopicName(TransferEventType.POSITION, TransferEventAction.PREPARE),
+      config: Utility.getKafkaConfig(Utility.ENUMS.CONSUMER, TransferEventType.TRANSFER.toUpperCase(), TransferEventAction.POSITION.toUpperCase())
     }
-  } catch (error) {
-    Logger.error(error)
-    throw error
+    positionHandler.config.rdkafkaConf['client.id'] = positionHandler.topicName
```
Comment (rmothilal) [inline src/handlers/positions/handler.js]: this should not be overwritten. take the client.id from config and concat a uuid to it

File: src/handlers/transfers/handler.js [hunk 5]
```diff
@@ -188,7 +189,7 @@ const prepare = async (error, messages) => {
       }
       Logger.info('TransferService::prepare::validationPassed::create position topic')
       // position topic to be created and inserted here
-      await Utility.produceParticipantMessage(payload.payerFsp, TransferEventType.POSITION, TransferEventAction.PREPARE, message.value, Utility.ENUMS.STATE.SUCCESS)
+      await Utility.produceGeneralMessage(TransferEventType.POSITION, TransferEventAction.PREPARE, message.value, Utility.ENUMS.STATE.SUCCESS, headers['fspiop-destination'])
```
Comment (rmothilal) [inline src/handlers/transfers/handler.js]: create enum for 'fspiop-destination' as well as remaining headers

File: src/handlers/transfers/handler.js [hunk 6]
```diff
@@ -312,15 +314,15 @@ const fulfil = async (error, messages) => {
           if (!Kafka.Consumer.isConsumerAutoCommitEnabled(kafkaTopic)) {
             await consumer.commitMessageSync(message)
           }
-          await Utility.produceParticipantMessage(existingTransfer.payeeFsp, TransferEventType.POSITION, TransferEventAction.COMMIT, message.value, Utility.ENUMS.STATE.SUCCESS)
+          await Utility.produceGeneralMessage(TransferEventType.POSITION, TransferEventAction.COMMIT, message.value, Utility.ENUMS.STATE.SUCCESS, headers['fspiop-destination'])
```
Comment (rmothilal) [inline src/handlers/transfers/handler.js]: create enum for 'fspiop-destination' as well as remaining headers

File: src/handlers/transfers/handler.js [hunk 7]
```diff
@@ -312,15 +314,15 @@ const fulfil = async (error, messages) => {
...
           histTimerEnd({success: true, fspId: Config.INSTRUMENTATION_METRICS_LABELS.fspId})
           return true
         } else {
           await TransferService.reject(transferId, payload)
           if (!Kafka.Consumer.isConsumerAutoCommitEnabled(kafkaTopic)) {
             await consumer.commitMessageSync(message)
           }
-          await Utility.produceParticipantMessage(existingTransfer.payerFsp, TransferEventType.POSITION, TransferEventAction.REJECT, message.value, Utility.ENUMS.STATE.SUCCESS)
+          await Utility.produceGeneralMessage(TransferEventType.POSITION, TransferEventAction.REJECT, message.value, Utility.ENUMS.STATE.SUCCESS, headers['fspiop-destination'])
```
Comment (rmothilal) [inline src/handlers/transfers/handler.js]: create enum for 'fspiop-destination' as well as remaining headers

File: src/handlers/lib/utility.js [hunk 8]
```diff
@@ -399,10 +399,10 @@ const createParticipantTopicConf = (participantName, functionality, action, part
  *
  * @returns {object} - Returns newly created general topicConfig
  */
-const createGeneralTopicConf = (functionality, action, partition = 0, opaqueKey = 0) => {
+const createGeneralTopicConf = (functionality, action, partition = 0, opaqueKey = null, key = null) => {
```
Comment (mdebarros) [inline src/handlers/lib/utility.js]: Same comment applies here. Should have function rather defined as follows:
```JAVASCRIPT
const createGeneralTopicConf = (functionality, action, key = null, partition = null, opaqueKey = null) => {}
```

File: src/handlers/lib/utility.js [hunk 9]
```diff
@@ -426,15 +426,15 @@ const createGeneralTopicConf = (functionality, action, partition = 0, opaqueKey
...
+const produceGeneralMessage = async (functionality, action, message, state, key) => {
   let functionalityMapped = functionality
   let actionMapped = action
   if (Enum.topicMap[functionality] && Enum.topicMap[functionality][action]) {
     functionalityMapped = Enum.topicMap[functionality][action].functionality
     actionMapped = Enum.topicMap[functionality][action].action
   }
   return await Kafka.Producer.produceMessage(updateMessageProtocolMetadata(message, functionality, action, state),
-    createGeneralTopicConf(functionalityMapped, actionMapped),
+    createGeneralTopicConf(functionalityMapped, actionMapped, 0, null, key),
```
Comment (mdebarros) [inline src/handlers/lib/utility.js]: Should be changed according to the above comment.

**Pull request merged**